src/main.rs

```diff
@@ -0,0 +1,160 @@
+use std::{
+    collections::HashMap,
+    env, fs,
+    time::{Duration, Instant},
+};
+
+use serde::{Deserialize, Serialize};
+use spectre::{graph::AGraph, graph::Graph};
+
+#[derive(Default, Clone, Serialize, Deserialize)]
+pub struct Node {
+    ip: String,
+    betweenness: f64,
+    closeness: f64,
+    num_connections: usize,
+}
+
+#[derive(Default, Clone, Serialize, Deserialize)]
+pub struct CrunchyState {
+    agraph_length: u32,
+    elapsed: f64,
+    nodes: Vec<Node>,
+    min_betweenness: f64,
+    max_betweenness: f64,
+    min_closeness: f64,
+    max_closeness: f64,
+}
+
+#[allow(dead_code)]
+#[derive(Default, Clone, Deserialize)]
+pub struct NetworkSummary {
+    num_known_nodes: usize,
+    num_good_nodes: usize,
+    num_known_connections: usize,
+    num_versions: usize,
+    protocol_versions: HashMap<u32, usize>,
+    user_agents: HashMap<String, usize>,
+    crawler_runtime: Duration,
+    node_ips: Vec<String>,
+    agraph: AGraph,
+}
+
+#[allow(dead_code)]
+#[derive(Default, Clone, Deserialize)]
+pub struct JsonRpcResponse {
+    jsonrpc: String,
+    result: NetworkSummary,
+    id: usize,
+}
+
+#[derive(Default, Clone, Deserialize)]
+pub struct ResponseSample {
+    pub response: JsonRpcResponse,
+}
+
+pub fn load_response(filepath: &str) -> JsonRpcResponse {
+    let jstring = fs::read_to_string(filepath).unwrap();
```

Comment:
If the provided input file doesn't exist, will the program panic?
Maybe we should add a check for that somewhere - in the `main()` and print out a help message

In any case, it would be good to replace `unwrap` with an `expect` function

Reply:
done

Reply:
good to have `expect` there, but I still don't think we should crash everything...

check my example below to see what I mean - I created an `ls` clone (fake clone) named `lss` which kills the process in case of unexpected input:
```
➜  debug git:(master) ✗ lss /dir_does_not_exist/file_does_not_exist
thread 'main' panicked at 'file /dir_does_not_exist/file_does_not_exist not found', src/main.rs:3:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

➜  debug git:(master) ✗ ls /dir_does_not_exist/file_does_not_exist
ls: /dir_does_not_exist/file_does_not_exist: No such file or directory
```
`ls` has a much nicer output, no 😅